File: backend/app/schemas/target.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from .image import Image
from .software import Software

class TargetBase(BaseModel):
    name: str
    description: Optional[str] = None
    base_image_id: int
    dockerfile: Optional[str] = None
    optimized_dockerfile: Optional[str] = None

class TargetCreate(TargetBase):
    software_ids: List[int]

class TargetUpdate(TargetBase):
    name: Optional[str] = None
    description: Optional[str] = None
    base_image_id: Optional[int] = None
    software_ids: Optional[List[int]] = None
    dockerfile: Optional[str] = None
    optimized_dockerfile: Optional[str] = None

class Target(TargetBase):
    id: int
    status: str
    created_at: datetime
    created_by_id: int
    base_image: Optional[Image] = None
    software_list: List[Software] = []

    class Config:
        from_attributes = True 

File: frontend/src/types/target.ts
export interface Target {
  id: number
  name: string
  description: string
  dockerfile: string
  optimized_dockerfile: string | null
  base_image_id: number
  software_ids: number[]
  ports: number[]
  created_at: string
  created_by_id: number
  base_image?: {
    id: number
    name: string
    registry_path: string
    version: string
    architecture: string
  }
  software_list: Array<{
    id: number
    name: string
    version: string
    architecture: string
    install_command: string
    start_command: string[]
    ports: number[]
  }>
}

export interface TargetCreate {
  name: string
  description?: string
  dockerfile?: string
  optimized_dockerfile?: string
  base_image_id: number
  software_ids: number[]
  ports: number[]
}

export interface TargetUpdate {
  name?: string
  description?: string
  dockerfile?: string
  optimized_dockerfile?: string
  base_image_id?: number
  software_ids?: number[]
  ports?: number[]
} 